src/wasm/literal.cpp

```diff
@@ -33,6 +33,67 @@ Literal::Literal(const uint8_t init[16]) : type(Type::v128) {
   memcpy(&v128, init, 16);
 }
 
+Literal::Literal(const Literal& other) {
+  type = other.type;
+  switch (type.getSingle()) {
+    case Type::i32:
+    case Type::f32:
+      i32 = other.i32;
+      return;
+    case Type::i64:
+    case Type::f64:
+      i64 = other.i64;
+      return;
+    case Type::v128:
+      memcpy(&v128, other.v128, 16);
+      return;
+    case Type::funcref:
+      func = other.func;
+      return;
+    case Type::exnref:
+      new (&exn) auto(std::make_unique<ExceptionPackage>(*other.exn));
+      return;
+    case Type::none:
+    case Type::nullref:
+      return;
+    case Type::anyref:
+    case Type::unreachable:
+      WASM_UNREACHABLE("unexpected type");
+  }
+  WASM_UNREACHABLE("unexpected type");
+}
+
+Literal& Literal::operator=(const Literal& other) {
```

Comment:
Can one of these be rewritten in terms of the other to reduce duplication? I think the constructor could be `*this = other`, but I could be wrong.

Reply:
I think that works in this case. Done.